slicedimage/io.py

```diff
@@ -11,24 +11,44 @@
 from six.moves import urllib
 
 from slicedimage.urlpath import pathjoin, pathsplit
-from .backends import CachingBackend, DiskBackend, HttpBackend
+from .backends import CachingBackend, DiskBackend, HttpBackend, SIZE_LIMIT
 from ._collection import Collection
 from ._formats import ImageFormat
 from ._tile import Tile
 from ._tileset import TileSet
 
 
-def infer_backend(baseurl, allow_caching=True):
+def infer_backend(baseurl, backend_config=None):
     """
     Guess the backend based on the format of `baseurl`, the consistent part of the URL or file path.
+    The backend_config dictionary can contain flexible parameters for the different backends.
+
+    Caching parameters include:
+
+     - cache.enabled (default: true)
+     - cache.directory (default: ~/.starfish-cache)
```

Comment:
see https://github.com/spacetx/starfish/pull/789/files#r240808045

Reply:
If that's the case, I'll likely drop `cache.enabled` and that will be determined by whether or not a directory has been set.